### DISCUSSION TITLE
Third party packages

### DISCUSSION THREAD

**Discussion by TimGeyssens** [TimGeyssens/Umbraco-v8-Community-Website-StarterKit#5] [General]
Which packages would make sense to include in the base build?

- uSync
- Full Text Search
- Matryoshka
- Contentment
- ...

**Comment by NikRimington** [comment]
- Friendly Sitemap
- Friendly Robots.txt
- Slimsy 

We often forget about sitemaps / robots.txt files so if these are part of the starter then we don't need to worry about them :-)

**Reply by NikRimington** [reply]
I've never looked at Full Text Search (I keep meaning to) so don't have a view on that.
There might be another package soon relating to the new block editor that might be useful.

**Reply by CarlSargunar** [reply]
Generally I like to keep projects as lean as possible to not create too many dependancies. Whilst I love a lot of packages out there is there an argument for keeping a starter kit clean so it's as re-useable as possible?

**Reply by TimGeyssens** [reply]
ah yes, good thinking @NikRimington !

**Reply by TimGeyssens** [reply]
@CarlSargunar well the basics should be covered I think, with proven open source packages from "trusted" sources

**Comment by dawoe** [comment]
I would try to not depend on packages if possible.  This will create to many dependencies that can break.
Also I think the aim of a starterkit is try to learn people on how to do something in Umbraco in the context of a specific user case, not how to use a certain package  

So you should showcase how to do search for example using examine. But tell people there is a package that could speed up things.

**Reply by TimGeyssens** [reply]
I think a nice part about umbraco is the package ecosystem... if we do everything custom...

**Reply by dawoe** [reply]
It is a important part if you ask me. But the goal is hopefully to teach people how to do things in Umbraco. So if you will do everything with packages... what will they learn ?

**Reply by TimGeyssens** [reply]
not everything :) just the boring stuff or stuff that shows how some packages make complex stuff super easy...

**Reply by CarlSargunar** [reply]
Perhaps an approach would be on a per-case basis, if there's already a package that does something that we need we use it, but we decide on a per package basis. 

A lot of packages exist becuase the functionality isn't in core, but arguably should be. That may never get fixed, but a starter kit shouldn't be too opinionated, and leave some choice to the user on how to implement things

**Comment by prgriffithsdev** [comment]
I think packages are great and offer some really useful functionality that might not come out of the box with the core. Aside from the ones already mentioned I tend to frequently use 'PageNotFoundManager' (by yourself Tim) to manage my 404's and 'The Dashboard' (by Markus Johansson). The dashboard comes in really useful if there is ever an instance where an exception is thrown after content updates, you can log in to Umbraco and check what the user was doing. Helps to quickly identify issues and get them fixed. Ofc content updates shouldn't cause exceptions if your code is good ;), you never know though! 

https://our.umbraco.com/packages/backoffice-extensions/the-dashboard/
https://our.umbraco.com/packages/backoffice-extensions/umbraco-page-not-found-manager/

**Reply by TimGeyssens** [reply]
Yes the dashboard is great! Good idea!
